src/include/libminiasync/vdm.h

```diff
@@ -36,6 +37,7 @@ struct vdm_memcpy_data {
 	void *src;
 	size_t n;
 	vdm_cb_fn vdm_cb;
+	void *vdm_data;
```

Comment:
This should be moved to `struct vdm` - it's per-mover state. And you should call init in `vdm_new` and fini in `vdm_delete`